mungegithub/mungers/flakeyjob-reporter.go

```diff
@@ -0,0 +1,278 @@
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"k8s.io/test-infra/mungegithub/mungers/mungerutil"
+)
+
+// FlakeyJob is a struct that represents a single job and the flake data associated with it.
+// FlakeyJob implements the Issue interface so that it can be synced with github issues via the IssueCreator.
+type FlakeyJob struct {
+	// Name is the job's name.
+	Name string
+	// Consistency is the percentage of builds that passed.
+	Consistency *float32 `json:"consistency"`
+	// FlakeCount is the number of flakes.
+	FlakeCount *int `json:"flakes"`
+	// FlakeyTests is a map of test names to the number of times that test failed.
+	// Any test that failed at least once a day for the past week on this job is included.
+	FlakeyTests map[string]int `json:"flakiest"`
+	// testsSorted is a list of the FlakeyTests test names sorted by desc. number of flakes.
+	// This field is lazily populated and should be accessed via TestsSorted().
+	testsSorted []string
+
+	// reporter is a pointer to the FlakeyJobReporter that created this FlakeyJob.
+	reporter *FlakeyJobReporter
+}
+
+// FlakeyJobReporter is a munger that creates github issues for the flakiest kubernetes jobs.
+// The flakiest jobs are parsed from JSON generated by /test-infra/experiment/bigquery/flakes.sh
+type FlakeyJobReporter struct {
+	flakeyJobDataURL string
+	syncCount        int
+	syncFreq         time.Duration
+
+	nextSync time.Time
+
+	creator *IssueCreator
+}
+
+func init() {
+	RegisterMungerOrDie(&FlakeyJobReporter{})
+}
+
+// Name is the name of munger.
+func (fjr *FlakeyJobReporter) Name() string { return "flakeyjob-reporter" }
+
+// Munge is unused by this munger.
+func (fjr *FlakeyJobReporter) Munge(obj *github.MungeObject) {}
+
+// RequiredFeatures specifies the features required by the FlakeyJobReporter (none needed).
+func (fjr *FlakeyJobReporter) RequiredFeatures() []string { return []string{} }
+
+func (fjr *FlakeyJobReporter) Initialize(config *github.Config, features *features.Features) error {
+	fjr.nextSync = time.Now().Add(time.Duration(-1) * time.Minute)
+
+	for _, munger := range GetActiveMungers() {
+		if munger.Name() == IssueCreatorName {
+			fjr.creator = munger.(*IssueCreator)
+		}
+	}
+	if fjr.creator == nil {
+		return fmt.Errorf("FlakeyJobReporter couldn't find the %s among active mungers.", IssueCreatorName)
+	}
+	return nil
+}
+
+// AddFlags specifies flags to add to the cobra `cmd`.
+func (fjr *FlakeyJobReporter) AddFlags(cmd *cobra.Command, config *github.Config) {
+	cmd.Flags().StringVar(&fjr.flakeyJobDataURL, "flakeyjob-data-url", "https://storage.googleapis.com/k8s-metrics/flakes-latest.json", "The url where flakey job JSON data can be found.")
+	cmd.Flags().IntVar(&fjr.syncCount, "flakeyjob-sync-count", 3, "The number of flakey jobs to try to sync to github.")
```

Comment:
Right, separating flakyjob flags from triage flags makes sense.

I don't know why we need to further hiearchically separate out flakyjob-data flags from flakyjob-sync flags. Alternatively I would expect it to either be `flaky-job-data-url` and `flaky-job-sync-count` or else `flakyjob-dataurl` and `flakyjob-synccount`.

However I personally feel like --count, --url, --freq are sufficiently verbose here,